connector/src/yang/connector/grpc/__init__.py

```diff
@@ -75,7 +76,9 @@ def __init__(self, *args, **kwargs):
         dev_args = self.connection_info
         self.log = log
         self.log.setLevel(logging.INFO)
-
+        self.proxy = dev_args.get('sshtunnel', {}).get('host')
+        self.source_address = kwargs.get('source_address')
+        self.vrf = kwargs.get('source_vrf') or self.device.management.get('vrf')
```

Comment:
source_vrf could be None or empty string, this would use management VRF. If the use does not want to use a VRF, this would not work. Please update logic to allow user to specify VRF, e.g.

```suggestion
        if 'source_vrf' in kwargs:
           self.vrf = kwargs.get('source_vrf')
        else:
           self.vrf = self.device.management.get('vrf')      
```